client/src/components/JobSections/JobDetails/index.tsx

```diff
@@ -0,0 +1,212 @@
+                    new_data.salary = parseInt(event.target.value);
+                    break;
+                case "url-value":
+                    new_data.jobPostUrl = event.target.value;
+                    break;
+                case "description-value":
+                    new_data.description = event.target.value;
+                    break;
+                default:
+                    break;
+            }
+            setData(new_data);
+        }
+    }
+
+    /**
+     * Saves locally typed information
+     */
+    const handleSave = () => {
+        if (data) {
+            // TODO: api call to save data
+            props.update(data);
+            setChanged(false);
+        }
+    }
+
+    if(!data) {
+        return <div>Loading...</div>;
+    } else {
+        return (
+            <Grid container spacing={{ xs: 2, md: 3 }} 
+                    columns={{ xs: 4, sm: 8, md: 12 }}
+                    sx={{ mx: 5 }}>
+                <Grid item >
+                    {data.company || data.company === "" ?
+                        <TextField id="company-value"
+                        variant="outlined" label="Company" size="small"
+                        onChange={handleTextChange}
+                        value={data.company} /> : <div></div>
+                    }
+                </Grid>
+                <Grid item >
+                    {data.position || data.position === "" ?
+                        <TextField id="position-value"
+                        variant="outlined" label="Job Title" size="small"
+                        onChange={handleTextChange}
+                        defaultValue={data.position} /> : <div></div>
+                    }
+                </Grid>
+                <Grid item >
+                    {data.jobPostUrl || data.jobPostUrl === "" ?
+                        <TextField id="url-value"
+                        variant="outlined" label="Job Post URL" size="small"
+                        onChange={handleTextChange}
+                        defaultValue={data.jobPostUrl} /> : <div></div>
+                    }
+                </Grid>
+                <Grid item >
+                    {data.location || data.location === "" ?
+                        <TextField id="location-value"
+                        variant="outlined" label="Location" size="small"
+                        onChange={handleTextChange}
+                        defaultValue={data.location} /> : <div></div>
+                    }
+                </Grid>
+                <Grid item >
+                    {data.salary ?
+                        <TextField id="salary-value"
+                        variant="outlined" label="Salary" size="small"
+                        onChange={handleTextChange}
+                        InputProps={{
+                            startAdornment: <InputAdornment position="start">$
+                                            </InputAdornment>
+                          }}
```

Comment:
Is it possible for you to fix this form field size? because of the startAdornment it makes it so the salary field box is longer than the rest of the other fields

Reply:
Honestly, did not even realize it before you mentioned it. Tbh, I would have to set a bunch of widths for every single one of my items (which may also introduce responsiveness issues to resolve) to solve this rather than letting them be as is. Not a priority, but we can keep it in mind at the end if there's little pieces we want to clean up. Thank you for pointing it out

Reply:
I do think it is possible because even when you resize the field, the box stays the same but its location may move (e.g. wrap)

Reply:
Yes, it is a possible fix, but I think an unnecessary one when looking at the effort to give consist widths and keep it responsive (afaik)